R/logistics_helper.R

```diff
@@ -0,0 +1,72 @@
+## log-likelihood
+logit_loglik <- function(beta, X, Y){
+  p <- 1 / (1 + exp(-X %*% beta))
+  l <- -(t(Y) %*% log(p) + t(1 - Y) %*% log(1 - p))
+  return(l)
+}
+
+
+## gradient
+logit_gradient <- function(beta, X, Y){
+  y_est <- 1 / (1 + exp(-X %*% beta))
+  gr = as.vector(t(X) %*% (y_est - Y))
+  return(gr)
+}
+
+## BFGS
+BFGS_finder_logit <- function(design, outcome, method){
+  init_coef <- rep(0, ncol(design))
+
+  obj_fn <- function(coef) {
+    logit_loglik(coef, design, outcome)
+  }
+
+  obj_grad <- function(coef){
+    logit_gradient(coef, design, outcome)
+  }
+
+  BFGS <- stats::optim(init_coef,
+                       obj_fn, obj_grad,
+                       method = method,
+                       control = list(fnscale = -1))
+  return(BFGS$par)
+}
+
+
+## Hessian matrix
+hessian <- function(beta, X, Y){
+  P <- 1 / (1 + exp(-X %*% beta))
```

Comment:
-1

Be consistent with naming and try not to use capital letters. This is `p` elsewhere. Similarly for function name, everything else is `logit_loglik/grad`, but here you just have `hessian`